Source/OnboardingContentViewController.m

```diff
@@ -187,10 +223,32 @@ - (void)generateView {
     CGFloat horizontalCenter = viewWidth / 2;
     CGFloat contentWidth = viewWidth * kContentWidthMultiplier;
     
-    // create the image view with the appropriate image, size, and center in on screen
-    _imageView = [[UIImageView alloc] initWithImage:_image];
-    [_imageView setFrame:CGRectMake(horizontalCenter - (self.iconWidth / 2), self.topPadding, self.iconWidth, self.iconHeight)];
-    [self.view addSubview:_imageView];
+    if (_image) {
+        
+        // create the image view with the appropriate image, size, and center in on screen
+        _imageView = [[UIImageView alloc] initWithImage:_image];
+        [_imageView setFrame:CGRectMake(horizontalCenter - (self.iconWidth / 2), self.topPadding, self.iconWidth, self.iconHeight)];
+        [self.view addSubview:_imageView];
+    
+    } else if (self.videoURL) {
+        
+        // or if we have a video create and configure the video player controller
+        self.moviePlayerController = [MPMoviePlayerController new];
+        self.moviePlayerController.contentURL = self.videoURL;
+        self.moviePlayerController.view.frame = self.view.frame;
+        self.moviePlayerController.repeatMode = MPMovieRepeatModeOne;
+        self.moviePlayerController.controlStyle = MPMovieControlStyleNone;
+        
+        UIImageView *thumbnailImageView = [[UIImageView alloc] initWithImage:[self thumbnailImageForVideo:self.videoURL]];
+        thumbnailImageView.frame = self.view.frame;
+        thumbnailImageView.contentMode = UIViewContentModeScaleAspectFit;
+        
+        [self.moviePlayerController.backgroundView addSubview:thumbnailImageView];
+        
+        [self.view addSubview:self.moviePlayerController.view];
+        
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleAppEnteredForeground) name:UIApplicationWillEnterForegroundNotification object:nil];
```

Comment:
Yes, I actually missed that, I added the ```dealloc``` with the ```removeObserver:``` call in ```OnboardingContentViewController``` and moved the ```addObserver``` to the init to be sure that the observer exist when the remove gets called.
Also added the  ```removeObserver ``` on  ```OnboardingViewController ``` cause didn't have it either.
I Just pushed the changes.